### ISSUE TITLE
strange work "text_encoder"

### ISSUE THREAD

**Issue by Azemar** [TheLastBen/fast-stable-diffusion#596]
Yesterday I started training the model for 3000 steps with intermediate models saved.
As a result, I got an intermediate model for 2500 steps and a final model for 3000.

Today I started training the model again with exactly the same parameters, but the final model was 2500 steps long.

Now I am comparing "intermediate at 2500" and "final at 2500" and I understand that for some reason they give different results.
I think it's about "text_encoder", which only works for the final model, without affecting the intermediate ones.

I also noticed such a bug: in the "Train" block with "resolution:" = 704, I get an error about the lack of memory in VRAM, but if I check the "Save_Checkpoint_Every_n_Steps:" checkbox, then there will be no error and the training will start.

I would also recommend that you automatically generate a .txt file in the "Session" folder with all the parameters that were used to generate the model.

**Comment by Azemar** [comment]
@TheLastBen

**Comment by TheLastBen** [comment]
the text encoder (once frozen) should be included in the intermediary checkpoints.

for the VRAM issue, "fp32" takes a bit more VRAM, if you want to use higher resolution pictures, keep it fp16 and check the box "reduce vram usage"

I'll see what I can do about a settings txt file

**Comment by Azemar** [comment]
> the text encoder (once frozen) should be included in the intermediary checkpoints.
> 
> for the VRAM issue, "fp32" takes a bit more VRAM, if you want to use higher resolution pictures, keep it fp16 and check the box "reduce vram usage"
> 
> I'll see what I can do about a settings txt file
"Reduce_memory_usage: on" and fp16 did not change the situation in any way - the same error.

There's also a new error when saving intermediate steps is enabled - Keyword arguments {'requires_safety_checker': False} not recognized.
![Скриншот_2022_11_27_16_06_16_888](https://user-images.githubusercontent.com/60752030/204136818-b57b1241-e825-4f86-965a-83118ef38e32.png)

**Comment by TheLastBen** [comment]
the V2 text_encoder is almost 3 times the size of the V1.5, so it takes more VRAM
for the warning, just ignore it, it doesn't break anything.

**Comment by Azemar** [comment]
> the V2 text_encoder is almost 3 times the size of the V1.5, so it takes more VRAM for the warning, just ignore it, it doesn't break anything.
Well, why then are intermediate models not generated?

**Comment by TheLastBen** [comment]
it writes : "killed", that means not enough RAM

**Comment by Azemar** [comment]
> it writes : "killed", that means not enough RAM, using fp16 ?

I am using fp32.
![Скриншот_2022_11_27_17_42_45_181](https://user-images.githubusercontent.com/60752030/204141271-9635851d-4dc4-4c33-8b88-442540ba7aa1.png)

**Comment by TheLastBen** [comment]
check the box fp16 or if you're using pro, set the vm shape to high-ram

**Comment by Azemar** [comment]
> check the box fp16 or if you're using pro, set the vm shape to high-ram

error
![Скриншот_2022_11_27_18_04_35_724](https://user-images.githubusercontent.com/60752030/204142251-9ee35121-1f6f-457d-be59-331aad836517.png)
![Скриншот_2022_11_27_18_05_28_582](https://user-images.githubusercontent.com/60752030/204142254-ce1041c6-5668-4ed0-929a-6948ec633a46.png)

**Comment by TheLastBen** [comment]
What resolution are you using ?

**Comment by Azemar** [comment]
> What resolution are you using ?

Same - 704

**Comment by TheLastBen** [comment]
the V2 has a bigger text_encoder, so it eats more VRAM

**Comment by Azemar** [comment]
At Resolution = 576, intermediate models are killed, and at 512 there is an error in general
![Скриншот_2022_11_28_20_20_44_899](https://user-images.githubusercontent.com/60752030/204341122-d8636011-ba8d-4ad1-af78-7d850afb8bff.png)

**Comment by TheLastBen** [comment]
it's fixed now for training on 512 images, update the colab

**Comment by Azemar** [comment]
> it's fixed now for training on 512 images, update the colab

Now both V2 models work at Resolution=512, but intermediate models are still "Killed". Sadly.

I also noticed a small bug - if you forcibly complete the download of instance images, the process will not stop, but will restart and continue downloading.
![Скриншот_2022_11_29_02_40_22_807](https://user-images.githubusercontent.com/60752030/204407472-5b5a8b5b-efa1-4699-898c-483f0014d404.png)

**Comment by TheLastBen** [comment]
That's how colab operates when it comes to a loop, only restarting the runtime will stop the process.

**Comment by Azemar** [comment]
> That's how colab operates when it comes to a loop, only restarting the runtime will stop the process.

Question about learning in multiple passes.
I have 250 images, if it takes 200 steps to train one image, then what is the best way to train:
-1- take 25 images, train for 5000 steps, and then use the trained model as a base for the next 25?
-2- take all 250 images at once, train for 5000 steps, and use the resulting model as a base 9 more times?

Due to a limitation in google colab compute blocks, I can only train for 5000 steps.

**Comment by TheLastBen** [comment]
train on all 250 images, set the steps to 4000 for example, then later you can resume the session and continue training.

**Comment by TheLastBen** [comment]
yes, share the session's folder in your gdrive, then paste the link in the session cell, and everything will load automatically, make sure you check the box "resume training" in the cell below.

**Comment by Azemar** [comment]
> yes, share the session's folder in your gdrive, then paste the link in the session cell, and everything will load automatically, make sure you check the box "resume training" in the cell below to continue training

why sometimes the message "Freezing the text_encoder" pops up?
![Скриншот_2022_11_30_14_08_33_692](https://user-images.githubusercontent.com/60752030/204780839-468ad06e-6f56-449e-8d93-fb1a692c267a.png)

**Comment by TheLastBen** [comment]
if you set the text encoder to 2%, it will freeze at 2%

**Comment by Azemar** [comment]
> if you set the text encoder to 2%, it will freeze at 2%

That is, the minimum allowable value for a text encoder is greater than 2%?

I tried to use a session from another account, but during training I get an error:
![Скриншот_2022_11_30_18_33_00_53](https://user-images.githubusercontent.com/60752030/204841177-091fda3a-5ef1-4345-b598-85c1e7487066.png)
![Скриншот_2022_11_30_18_33_07_539](https://user-images.githubusercontent.com/60752030/204841182-9201d900-4e01-4b3f-9e04-cd4c68db1574.png)
![Скриншот_2022_11_30_18_33_33_766](https://user-images.githubusercontent.com/60752030/204841190-b2627d0b-15c5-4c24-b5ce-875e1658ad85.png)
![Скриншот_2022_11_30_18_33_44_234](https://user-images.githubusercontent.com/60752030/204841200-3fd405b5-e738-41a8-bb74-6ef7edc5c981.png)
But there are images and a zip archive in the session folder itself.

**Comment by TheLastBen** [comment]
There should be a folder name "instance_images" that contains the input images, the zip file is there only in case you load the session from a different account.

**Comment by Azemar** [comment]
> why did you use that path? you use that only if you're loading a session from a different drive, and you use a shared link, not a path

I use this path because I created a folder shortcut

**Comment by TheLastBen** [comment]
you can't use a path in the session link

**Comment by Azemar** [comment]
> you can't use a path in the session link

After training the model, the "tokenizer" folder appeared. Is that how it should be?
![Скриншот_2022_11_30_22_56_45_65](https://user-images.githubusercontent.com/60752030/204896033-7a60f537-9d8f-494e-9198-15d1fc619e04.png)
![Скриншот_2022_11_30_22_57_06_650](https://user-images.githubusercontent.com/60752030/204896040-9a03de40-6e03-47fd-a043-a94d130eb0e9.png)
![Скриншот_2022_11_30_22_58_50_233](https://user-images.githubusercontent.com/60752030/204896178-6b224022-e6eb-467c-9043-2ab375ca2220.png)

I also saved intermediate models every 500 steps, and at step 3999 a model is generated with the name "ByNispArt_step_4000.ckpt", and after a minute (as I understand it, the final one) ByNispArt.ckpt. If there are no differences between them, then I think it makes no sense to generate the same thing twice.

**Comment by Azemar** [comment]
> yes, share the session's folder in your gdrive, then paste the link in the session cell, and everything will load automatically, make sure you check the box "resume training" in the cell below to continue training

I tried to continue training, but for some reason the program does not find the model
![Скриншот_2022_12_03_12_43_31_713](https://user-images.githubusercontent.com/60752030/205434636-d6ea7c23-29f3-45c4-b967-f61e6d2abed0.png)
![Скриншот_2022_12_03_12_43_41_646](https://user-images.githubusercontent.com/60752030/205434642-9f6a256a-df7c-4102-9c96-0907c1c1049d.png)
![Скриншот_2022_12_03_12_43_48_579](https://user-images.githubusercontent.com/60752030/205434644-fee15be8-d9b1-4696-bbef-9ab850f3293f.png)
